File: all3_trend.m
function [Tresult_MK,Tresult_LMSlog,Tresult_LMSlin]=all3_trend(data,param, inst, station, resolution, varargin)

% do the MK, LM and MK trend and save the results and figures
%data: should be as time table

%Dependency: call seasonalKendall_main_D & trend_LMS_D

Tresult_MK=seasonalKendall_main_D(data,param, inst, station, resolution,varargin{:});
%Tresult_GSMm=bootstrapconfidence_seas_D(data, param,inst, station, 'lin','granu','monthly',varargin{:});
%Tresult_GSMd=bootstrapconfidence_seas_D(data, param,inst, station, 'lin','granu','daily',varargin{:});
Tresult_LMSlog=trend_LMS_D(data, param,inst, station, 'log',varargin{1:end-2},'fig',0);
%Tresult_LMSlog2=trend_LMS_deseasonFit(data, param,inst, station, 'log',varargin{:});

Tresult_LMSlin=trend_LMS_D(data, param,inst, station, 'lin',varargin{:});
%Tresult_LMSlin2=trend_LMS_deseasonFit(data, param,inst, station, 'lin',varargin{:});
%Tresult=[Tresult_MK ;Tresult_GSMd;Tresult_LMS;Tresult_LMS2];


% old version: [Tresult_MK,Tresult_GSMd,Tresult_LMSlog,Tresult_LMSlin,Tresult_LMSlog2,Tresult_LMSlin2]=all3_trend(data,param, inst, station, resolution, varargin)

File: all3_trend.py
from trend_LMS_D import trend_LMS_D
from seasonalKendall_main_D import seasonalKendall_main_D

def all3_trend(data,param,inst,station,resolution,**kwargs):
    """ do the MK, LM and MK trend and save the results and figures
    data: should be as time table"""

    #Dependency: call seasonalKendall_main_D & trend_LMS_D
    # Mann-Kendall seasonal trend
    Tresult_MK = seasonalKendall_main_D(data,param,inst,station,resolution,**kwargs)

    # LMS logarithmic
    kwargs_log = kwargs.copy()
    kwargs_log["fig"] = False
    Tresult_LMSlog, _, _  = trend_LMS_D(data,param,inst,station,"log",**kwargs_log)

    # LMS linear
    Tresult_LMSlin, _, _  = trend_LMS_D(data,param,inst,station,"lin",**kwargs)

    return (Tresult_MK,Tresult_LMSlog,Tresult_LMSlin)